rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -0,0 +1,524 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="hibernate4" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides analysis for migration to Hibernate 5.x.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.4.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.4.0.Final" />
+        </dependencies>
+        <sourceTechnology id="hibernate" versionRange="[4,)" />
+        <targetTechnology id="hibernate" versionRange="[5,)" />
+        <targetTechnology id="eap" versionRange="[7,)" />
+        <tag>reviewed-2016-02-25</tag>
+    </metadata>
+    <rules>
+        <rule id="hibernate4-00001">
+            <when>
+                <javaclass references="org.hibernate.cfg.Configuration.addFile{*}">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Method org.hibernate.cfg.Configuration.addFile is deprecated in Hibernate 5" effort="1"
+                    severity="mandatory">
+                    <message>Use `org.hibernate.boot.MetadataSources#addFile` instead.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#re-purposing-of-configuration"
```

Comment:
This is a bit of chicken/egg problem, IMHO. I'd personally prefer if all the migration rules linked to EAP 7 Migration Guide (because it's more "official" and goes through QA, so its content is IMHO slightly better than the WildFly wiki), but how can we do that if the latest improvements in the Migration Guide are not yet published and the Migration Guide for 7.0.0 will be on a different URL than the Migration Guide for 7.0 Beta?

I don't know.